designs/cloud_formation_api_gateway_support.md

```diff
@@ -0,0 +1,377 @@
+
+```yaml
+ServerlessRestApi:
+Type: 'AWS::ApiGateway::RestApi'
+Properties:
+  Body:
+    basePath : /pete
+    paths:
+      /hello:
+        get:
+          x-amazon-apigateway-integration:
+            httpMethod: POST
+            type: aws_proxy
+            uri: !Sub >-
+              arn:aws:apigateway:${AWS::Region}:lambda:path/2015-03-31/functions/${HelloWorldFunction.Arn}/invocations
+          responses: {}  
+```
+
+Swagger can also be inlined using the FN::Include Macro. This should also be supported while defining the CloudFormation template.
+****Feature* #2*: Using a combination of AWS::ApiGateway::Resource with ApiGateway::Methods.**
+
+Although this approach is less common and more verbose, it is still used by some people while defining their resources. Tools 
+such as aws-cdk currently generate Resource and Methods ApiGateway CloudFormation Resources in their yaml instead of swagger. 
+
+```yaml
+UsersResource:
+  Type: AWS::ApiGateway::Resource
+  Properties:
+    RestApiId:!Ref RestApi
+    ParentId:
+      Fn::GetAtt:
+      - RestApi
+      - RootResourceId
+    PathPart: user
+UsersGet:
+  Type: AWS::ApiGateway::Method
+  Properties:
+    ResourceId: !Ref ApiGatewayResource
+    RestApiId: !Ref ApiGatewayRestApi
+    Integration:
+      Type: AWS
+      IntegrationHttpMethod: POST
+      Uri:
+        Fn::Join:
+        - ''
+        - - 'arn:aws:apigateway:'
+          - Ref: AWS::Region
+          - ":lambda:path/2015-03-31/functions/"
+          - Fn::GetAtt:
+            - Lambda
+            - Arn
+          - "/invocations"
+      IntegrationResponses: []
+    MethodResponses:
+    - ResponseModels:
+        application/json:
+          Ref: UsersModel
+      ResponseParameters:
+        method.response.header.Link: true
+      StatusCode: 200
+   ApiGatewayMethod:
+    Type: AWS::ApiGateway::Method
+    Properties:
+      HttpMethod: POST
+      Integration:
+        ConnectionType: INTERNET
+        IntegrationResponses:
+          - ResponseTemplates:
+              application/json: "{\"message\": \"OK\"}"
+            StatusCode: 200
+        RequestTemplates:
+          applicationjson: "{\"statusCode\": $input.json('$.statusCode'), \"message\": $input.json('$.message')}"
+
+```
+
+The SAM template requires people to use only a single RestApi to define all of their resources.
+
+### **AWS::ApiGateway::Stages**
+
+The stage allows for defining environments for different parts of the pipeline in CloudFormation. Since a single stage is attached to a deployment, the behavior of multiple deployment/stages locally is currently undefined, which is similar to the current support or syntax in the SAM template. 
```

Comment:
"the behavior of multiple deployment/stages locally is currently undefined, which is similar to the current support or syntax in the SAM template."

Kind of. Since AWS::Serverless::Api only allows one stage, we never have to pick if other stages are in the template.